src/Window.cpp

```diff
@@ -1109,7 +1110,14 @@ QString Window::executeCommand(QString command) {
         return boolToString(wallLeft());
     }
     else if (function == "moveForward") {
-        bool success = moveForward();
+        bool success;
+        if (tokens.size() == 2) {
+            int distance = tokens.at(1).toInt();
+            success = moveForward(distance);
+        }
+        else {
+            success = moveForward();
+        }
```

Comment:
Rather than optionally providing a distance to the `moveForward()` argument, just use a default value of `1` if a second argument isn't provided. Then we don't need a default argument in the `moveForward()` function:

```cpp
int distance = 1;
if (tokens.size() == 2) {
    distance = tokens.at(1).toInt();
}
bool success = moveForward(distance);